### PR TITLE
feat: remove bank multisend msg

### PR DESCRIPTION
**Pull request by sunnya97**
Closes: https://github.com/cosmos/cosmos-sdk/issues/12397

## What is the purpose of the change

This PR remove the Bank MultiSend msg.  I believe it was created before MultiMsg txs were created, and it serves mostly duplicate usage.

Part of the motive for this is that we need to add a hook to the Send functionality of the bank module, but the MsgMultiSend bypasses it because it doesn't use the SendCoins function of the Bank keeper and instead uses a SubtractCoins and AddCoins.

## Brief Changelog

 
  - Removes all usage of MultiSend and InputOutputCoins

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/bank/types/errors.go [hunk 1]
```diff
@@ -6,9 +6,9 @@ import (
 
 // x/bank module sentinel errors
 var (
-	ErrNoInputs              = sdkerrors.Register(ModuleName, 2, "no inputs to send transaction")
-	ErrNoOutputs             = sdkerrors.Register(ModuleName, 3, "no outputs to send transaction")
-	ErrInputOutputMismatch   = sdkerrors.Register(ModuleName, 4, "sum inputs != sum outputs")
+	// ErrNoInputs              = sdkerrors.Register(ModuleName, 2, "no inputs to send transaction")
+	// ErrNoOutputs             = sdkerrors.Register(ModuleName, 3, "no outputs to send transaction")
+	// ErrInputOutputMismatch   = sdkerrors.Register(ModuleName, 4, "sum inputs != sum outputs")
```
Comment (alexanderbez) [inline x/bank/types/errors.go]: Can we remove these?

**Review by alexanderbez — APPROVED** [review]

File: x/bank/types/errors.go (hunk 1 above)
Comment (sunnya97) [reply x/bank/types/errors.go]: I kept them as a way to reserve the error codes so they don't get accidentally recycled.  I can just switch it to a comment saying the error codes are reserved?

Comment (alexanderbez) [reply x/bank/types/errors.go]: Yeah I was gonna suggest that 👍

**Review by mattverse — APPROVED** [review]
LGTM! 

there's deprecated comment for multisend left in `interface_registry.go` which should be preferably deleted, but otherwise good to hit merge!

**Comment by mattverse** [comment]
nm! took care of the deprecated comment!

**Pull request merged**